sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -676,7 +676,7 @@ static void rtt_cb(void *arg)
 {
     msg_t msg;
 
-    msg.content.value = ((uint32_t) arg) & 0xffff;
+    msg.content.value = ((uintptr_t) arg) & 0xffff;
```

Comment:
Not sure if an outer cast might be necessary as well:

```suggestion
    msg.content.value = (uint32_t)(((uintptr_t) arg) & 0xffff);
```

(my intuition says it shouldn't as `sizeof(uintptr_t) <= sizeof(uint32_t)` should always be true. So let's wait for Murdock to give the verdict on that.

Reply:
It wouldn't be true for 64 bit architectures.
I suppose casting to `uint16_t` is the better option then.

Reply:
Now that you say start talking about wider pointers than `uint32_t` ... `msg.content.value` is in a `union` with `msg.content.ptr`. Of course there is the very unlikely case that `arg` is not in the range of `uint16_t`, by why not just do

```suggestion
    msg.content.ptr = arg;
```

?

Reply:
(of course there should be a comment explaining what's going on there)

Reply:
Now that's something I'd like to avoid.
It implicitly makes assumptions on how the memory is layed out in the union if the size of `uint32_t` and `void*` doesn't match.
I wouldn't intuitively know if this works on all architectures - let's rather be explicit.

Reply:
Ok, then go for the outer cast option.